lib/archethic/mining/validation_context.ex

```diff
@@ -800,30 +798,31 @@ defmodule Archethic.Mining.ValidationContext do
 
   defp get_ledger_operations(
          %__MODULE__{
-           transaction: tx,
+           transaction: tx = %Transaction{address: address, type: tx_type},
            unspent_outputs: unspent_outputs,
            resolved_addresses: resolved_addresses
          },
          fee,
          validation_time,
          encoded_state
        ) do
-    resolved_movements =
-      tx
-      |> Transaction.get_movements()
-      |> TransactionMovement.resolve_movements(resolved_addresses)
-
-    %LedgerOperations{
-      fee: fee,
-      transaction_movements: resolved_movements,
-      tokens_to_mint: LedgerOperations.get_utxos_from_transaction(tx, validation_time),
-      encoded_state: encoded_state
-    }
-    |> LedgerOperations.consume_inputs(
-      tx.address,
-      unspent_outputs,
-      validation_time |> DateTime.truncate(:millisecond)
-    )
+    movements = Transaction.get_movements(tx)
+
+    {sufficient_funds?, ops} =
+      LedgerOperations.consume_inputs(
+        %LedgerOperations{fee: fee},
+        address,
+        unspent_outputs,
+        movements,
+        LedgerOperations.get_utxos_from_transaction(tx, validation_time),
```

Comment:
I'm wondering if we should not just concat unspent_outputs with the new unspent outputs  (tokens_to_mint) at first.
This would simplify the code of the `consume inputs` fn , because at the end `tokens_to_mint` is only used to be concatenated into the list of inputs.

Reply:
After AEIP21, the new tokens will still be reaming in the UTXO and only be in the consumed inputs if the token define minted transfers. So I don't see yet the benefits to keep them separately

Reply:
A transaction could consume inputs not yet created ?

Reply:
Yeah..maybe not.
It might be strange evenif in a pure a UTXO model , an output should come from an input .

Reply:
In case of token minting, the output could not come from inputs since the transaction is creating new things so I guess it's valid

Reply:
Token minting is a particular use case for UTXO, not initial thought from it.
I think both could still be valid depending of people point of view but for simplification we can omit it, particularly than after AEIP21 
`new_token_unspent_outputs` might not be used and we would need identification of the `tokens_to_mint`